content/c-sharp/concepts/deque/deque.md

```diff
@@ -0,0 +1,109 @@
+---
+Title: 'Deque'
+Description: 'A deque is a type of data structure that allows insert and delete elements at both ends.'
+Subjects:
+  - 'Computer Science'
+  - 'Code Foundations'
+Tags:
+  - 'Deques'
+  - 'Data Structures'
+  - 'Data Types'
+CatalogContent:
+  - 'learn-c-sharp'
+  - 'paths/computer-science'
+---
+
+A **deque** is a data structure that allows elements to be added or removed from both ends, making it more versatile than a traditional queue or stack. In `C#` there is no build-in `deque` but it can be impleted using `LinkedList` and `List` classes `C#`.
```

Comment:
```suggestion
A **deque** is a data structure that allows elements to be added or removed from both ends, making it more versatile than a traditional queue or stack. In `C#`, there is no built-in `deque` but it can be impleted using `LinkedList` and `List` classes.
```
Corrected some punctuation marks.